File: RP04_EstruturaDados/Exercicio.cpp
#include <iostream>
#include <vector>
using namespace std;

int main() {

    // ARRAY FIXO

    string nomesArray[3] = {"Chocolate", "Morango", "Baunilha"};
    int qtdArray[3] = {10, 3, 0};

    cout << "=== ARRAY FIXO ===" << endl;

    for (int i = 0; i < 3; i++) {
        cout << nomesArray[i] << " - " << qtdArray[i] << endl;
    }

    // LISTA DINÂMICA

    vector<string> nomes = {"Chocolate", "Morango", "Baunilha"};
    vector<int> quantidades = {10, 3, 0};

    cout << "\n=== LISTA DINÂMICA ===" << endl;

    for (int i = 0; i < nomes.size(); i++) {
        int qtd = quantidades[i];
        string status;

        if (qtd == 0) {
            status = "Em falta";
        } else if (qtd < 5) {
            status = "Alerta";
        } else {
            status = "OK";
        }

        cout << "Produto: " << nomes[i] << endl;
        cout << "Quantidade: " << qtd << endl;
        cout << "Status: " << status << endl;
        cout << "------------------" << endl;
    }

    return 0;
}

File: RP04_EstruturaDados/Exercicio.java
import java.util.ArrayList;

public class Exercicio {
    public static void exercicio(String[] args) {

        // ARRAY FIXO

        String[] nomesArray = {"Chocolate", "Morango", "Baunilha"};
        int[] qtdArray = {10, 3, 0};

        System.out.println("=== ARRAY FIXO ===");
        for (int i = 0; i < nomesArray.length; i++) {
            System.out.println(nomesArray[i] + " - " + qtdArray[i]);
        }

        // LISTA DINÂMICA

        ArrayList<String> nomes = new ArrayList<>();
        ArrayList<Integer> quantidades = new ArrayList<>();

        nomes.add("Chocolate");
        quantidades.add(10);

        nomes.add("Morango");
        quantidades.add(3);

        nomes.add("Baunilha");
        quantidades.add(0);

        System.out.println("\n=== LISTA DINÂMICA ===");

        for (int i = 0; i < nomes.size(); i++) {
            int qtd = quantidades.get(i);
            String status;

            if (qtd == 0) {
                status = "Em falta";
            } else if (qtd < 5) {
                status = "Alerta";
            } else {
                status = "OK";
            }

            System.out.println("Produto: " + nomes.get(i));
            System.out.println("Quantidade: " + qtd);
            System.out.println("Status: " + status);
            System.out.println("------------------");
        }
    }
}